metabase/metabase
# Action Creator 3: Form Field Type Editing

## Description

- Adds types for `ActionFormSettings` and `FieldSettings`
- Adds FieldSettingsPopover to set `fieldType` and `inputType`
- Adds very basic/placeholdery input previews

## Screenshots
![Screen Shot 2022-08-30 at 3 24 09 PM](https://user-images.githubusercontent.com/30528226/187545976-25182337-f0b5-4d4f-a7b4-e006e94fcbcf.png)
![Screen Shot 2022-08-30 at 3 23 48 PM](https://user-images.githubusercontent.com/30528226/187545979-0293e7e2-bc00-4c3e-990b-3b165fdec7fa.png)

https://www.loom.com/share/6bc31779ca8546c69ebe115c6aa4ba2e

## Testing Steps

- create an action by hitting `new > action` and write some sql with {{ template_tags }}
- using the cog, open the field settings popover, and customize the input types, see that the input types persist when the popover closes
- save the action and see that the settings were sent tot he backend in the `visualization_settings` parameter
- if you can, find the action id for your newly created action, and open it again via the url `/action/:id` and see that the form settings persisted and loaded properly

## Out of scope

- to keep this PR reasonably sized, I tried not to be exhaustive with input types. I think there's some product/design work that needs to be done there, but I did try to create a reasonable structure to define the options for each field type
